patches/hideSubjectsWithNoGrades/script.js

```diff
@@ -0,0 +1,34 @@
+const SEMESTERS = 2;
+const renderVisibilityButtons = () => {
+    for (let i = 0; i < SEMESTERS; i++) {
+        hideEmptyFinalGradesInfo(i)
+    }
+}
+
+const getSemestersContainer = () => document.querySelector("section > section > .mobile__frame > .content-container").children
+
+const hideEmptyFinalGradesInfo = async (i) => {
+    await window.waitForRender(() => getSemestersContainer()[i].querySelector("article"), document.querySelector("section > section > .mobile__frame > .content-container"))
+    const container = getSemestersContainer()[i]
+    const subjects = container.querySelectorAll("article")
+    for (const subject of subjects) {
+        const finalGrades = document.querySelector(".tile__content:last-of-type")
```

Comment:
This should point to the `subject` not to whole document

```suggestion
        const finalGrades = subject.querySelector(".tile__content:last-of-type")
```